### DISCUSSION TITLE
Restricting fields that are returned from a query

### DISCUSSION THREAD

**Discussion by ccsv** [mirumee/ariadne#907] [Q&A]
Just want to ask if this is possible in the resolver.

Lets say I have an schema

```
type Client {
    id:  ID
    name: String!
    notes: String
   }
   
   type Query {
    Clients(id:ID, name:String, notes:String ): [Client]
   }
```
   
and a resolver
   
```
@query.field("Clients")
def resolve_Clients(*_, name=None, id=None,):
    if name:
        filter = Q(name__icontains=name)
        return Client.objects.filter(filter)
    if id:
        filter = Q(id__exact=id)
        return Client.objects.filter(filter)
    return Client.objects.all()   
```

My query looks like this (in javascript):

```
export const allClients = gql`
query Clients($id:ID, $name:String) {
  Clients(id:$id, name:$name) {
    id
    name
    notes
  }
}
`;
```

If for certain instances (like if I have a user in a different group) I want to restrict a field like `notes` would I have to rewrite another query and load or is there any other options in the resolver where I can place restrictions on what fields are returned?

**Comment by glennlaughlin** [comment]
https://ariadnegraphql.org/docs/schema-directives#applying-directives-to-schema-items ?

**Comment by rafalp** [comment]
Simplest solution is to return `None` from resolver, making field's value `null` in the client, then assuming in the client logic that `null` for `notes` means that you don't have permission to see them.
